File: feature-main/src/main/java/com/smedialink/feature_main/delegate/BackPressedDelegate.java
package com.smedialink.feature_main.delegate;

import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import com.smedialink.feature_main.R;

import javax.inject.Inject;

import home.oleg.placesnearme.core_presentation.base.BackHandler;

public class BackPressedDelegate implements OnBackPressListener {

    private final AppCompatActivity appCompatActivity;

    @Inject
    public BackPressedDelegate(AppCompatActivity appCompatActivity) {
        this.appCompatActivity = appCompatActivity;
    }

    @Override
    public boolean onBackPressed() {
        Fragment curFrag = appCompatActivity.getSupportFragmentManager().findFragmentById(R.id.container);
        boolean intercepted = false;

        if (curFrag instanceof BackHandler) {
            intercepted = ((BackHandler) curFrag).onBackPressed();
        }

       return intercepted;

    }

}

File: feature-main/src/main/java/com/smedialink/feature_main/delegate/BackPressedDelegate.kt
package com.smedialink.feature_main.delegate

import com.smedialink.feature_main.R

import javax.inject.Inject

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import home.oleg.placesnearme.core_presentation.base.BackHandler

class BackPressedDelegate @Inject
constructor(private val appCompatActivity: AppCompatActivity) : OnBackPressListener {

    override fun onBackPressed(): Boolean {
        val curFrag = appCompatActivity.supportFragmentManager.findFragmentById(R.id.container)
        var intercepted = false

        if (curFrag is BackHandler) {
            intercepted = (curFrag as BackHandler).onBackPressed()
        }

        return intercepted

    }

}
